### DISCUSSION TITLE
Apply filter on DatagridTemplateColumn

### DISCUSSION THREAD

**Discussion by Ja7373** [dotnet/DataGridExtensions#87] [Q&A] [answered]
Hello All,
I want to filter a DataGridTemplateColumn
How can I apply the filter on these columns?

```
<DataGridTemplateColumn Width="Auto">
      <DataGridTemplateColumn.Header>
        <TextBlock Text="Name" FontWeight="Bold" Margin="5 0"/>
      </DataGridTemplateColumn.Header>
      <DataGridTemplateColumn.CellTemplate>
        <DataTemplate>
          <TextBlock Text="{Binding Name}" Margin="5 0"/>
        </DataTemplate>
      </DataGridTemplateColumn.CellTemplate>
</DataGridTemplateColumn>
```
Thanks.

**Comment by tom-englert** [comment, marked as answer]
https://github.com/dotnet/DataGridExtensions/blob/master/src/DataGridExtensionsSample/Views/Customized2View.xaml#L188-L195

**Reply by Ja7373** [reply]
Thanks Tom.

**Reply by infinitesquirrel** [reply]
Hi @tom-englert, is the MultipleChoiceFilter also working with DataGridTemplateColumn?
I checked with the "Basic-Sample":
with the example implementation - a text column - the filter gets updated as the column value changes. 
https://github.com/dotnet/DataGridExtensions/blob/17a6ae30426bd0e13e3a44b45ef1719b8426c4ab/src/BasicSample/MainWindow.xaml#L20 
When I'm replacing this column / line with a template column according to the syntax as in the other example:
`          
          <DataGridTemplateColumn Header="Template" SortMemberPath="Column5" dgx:DataGridFilterColumn.Template="{StaticResource MultipleChoiceFilter}">
              <DataGridTemplateColumn.CellTemplate>
                  <DataTemplate DataType="{x:Type basicSample:DataItem}" >
                      <TextBox Margin="2" Text="{Binding Column5}"/>
                  </DataTemplate>
              </DataGridTemplateColumn.CellTemplate>
          </DataGridTemplateColumn>`
          
The Dropdown Values don't get updated when I'm changing values in the cell.
Am I missing some event bindings / etc that need to be made additionally for a template column?

**Reply by magauthier-liggo** [reply]
This comment would be great in Filter / Sort documentation. Thanks!
" A template column may contain everything, so we need to specify the SortMemberPath explicitly.."

**Reply by tom-englert** [reply]
Happy to accept your PR 😃
